kong/plugins/prometheus/exporter.lua

```diff
@@ -17,7 +12,7 @@ local prometheus
 local function init()
   local shm = "prometheus_metrics"
   if not ngx.shared.prometheus_metrics then
-    ngx_log(ERR, "prometheus: ngx shared dict 'prometheus_metrics' not found")
+    kong.log.ERR("prometheus: ngx shared dict 'prometheus_metrics' not found")
```

Comment:
`kong.log.ERR` is not a function, this will attempt to call `nil`. `kong.log.err` is the correct name. Was this change tested?

Reply:
There is no test for this change and is a difficult one to test since this condition doesn't occur in in Kong >= 0.14.0 as Kong injects the shm whenever it detects `prometheus` plugin is loaded at startup.

Reply:
By "this change", I am referring to this entire diff (any of the error logs produced in any of the unhappy paths touched by this patch). I already assume that this answer is no. What will we do to fix this patch and ship another release?

Reply:
You probably know but just want to note that the diff was tested against `next` branch of Kong's Git repo.

To fix this patch, I'll make another PR to fix the `kong.log.*` functions and will try to come up with tests for unhappy path on Line 85 of this file but I don't know how to test other unhappy paths.

Other unhappy paths are rare conditions that are hard to simulate, but happy to write those tests if it's possible to do those. Any pointers that I could refer?

Reply:
> To fix this patch, I'll make another PR to fix the `kong.log.*` functions

Sweet!

> I don't know how to test other unhappy paths.

We don't test _all_ unhappy paths (especially when they are tricky to catch). They often make a large part of the uncovered portion of many projects we develop or work with. Simply commenting out or tweaking an unhappy condition (like commenting out the surrounding branch) could also tell us if `kong.log.ERR` is offered as an alias to `kong.log.err` (spoiler alert: it isn't). Of course, if they aren't already, testing unhappy paths as part of CI would set us up a million times better off :+1: 

> Other unhappy paths are rare conditions that are hard to simulate, but happy to write those tests if it's possible to do those. Any pointers that I could refer?

I have tested race conditions with various tricks in the past (ngx.thread/sleep/semaphore/etc...) but they are probably not worth the time for such high-level utility like this plugin, unless you have lots of time on your hands :)